### PR TITLE
Add support for selecting GPU instance based on Kata pod annotations

### PR DESCRIPTION
**Pull request by bpradipt**
Currently support is added only for AWS provider. Couldn't find a way to retrieve GPU details of an instance in Azure.
Depends on https://github.com/kata-containers/kata-containers/pull/10453

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

File: src/cloud-providers/util.go [hunk 1]
```diff
@@ -102,6 +114,21 @@ func GetBestFitInstanceType(sortedInstanceTypeSpecList []InstanceTypeSpec, vcpus
 
 }
 
+// Implement the GetBestFitInstanceTypeWithGPU function
+func GetBestFitInstanceTypeWithGPU(sortedInstanceTypeSpecList []InstanceTypeSpec, gpus, vcpus, memory int64) (string, error) {
```
Comment (snir911) [inline src/cloud-providers/util.go:118]: Should the original GetBestFitInstanceType make sure 0 GPUs instance type are returned (in case gpuCount is 0)?

**Review by snir911 — APPROVED** [review]
LGTM, Thanks!, one minor comment,
Maybe worth pointing in the doc also to the [peerpod-ctrl resource management](https://github.com/confidential-containers/cloud-api-adaptor/tree/main/src/peerpod-ctrl) as "resource management" is ambiguous term in the peer-pods context..

**Comment by bpradipt** [comment]
The TestSshProxyReverseKBS is flaky. I'm merging this one while we fix the test case

**Pull request merged**